### PR TITLE
Fix scheduler search and styles

### PR DESCRIPTION
**Pull request by codyebberson**
* Scheduler currently broken in storybook due to missing `Slot-schedule` and `Slot-start` search parameters in storybook
* Ported styles over to Mantine/Emotion
* Cleaned up some dumb logic code

### CODE REVIEW

File: packages/core/src/match.ts [hunk 1]
```diff
@@ -69,11 +69,8 @@ function matchesReferenceFilter(resource: Resource, filter: Filter, searchParam:
       // We need to maintain this behavior for backwards compatibility
       match = references.some((reference) => reference?.endsWith('/' + filterValue));
     }
-    if (match && !negated) {
-      return true;
-    }
-    if (match && negated) {
-      return false;
+    if (match) {
+      return !negated;
```
Comment (codyebberson) [inline packages/core/src/match.ts:73]: This is logically equivalent -- I'm not sure why I didn't write it this way in the first place.

File: packages/generator/src/storybook.ts [hunk 2]
```diff
@@ -20,12 +20,11 @@ const resourceTypes = [
   'Bot',
   'Project',
   'Communication',
-  'SpecimenDefinition',
```
Comment (codyebberson) [inline packages/generator/src/storybook.ts:23]: @rahul1 - It looks like `SpecimenDefinition` and`RequestGroup` were added to `storybook.ts`, but not actually executed, so the definitions were not in `packages/mocks/structuredefinitions.json`.  If they are needed, then we can add these back.

**Review by reshmakh — APPROVED** [review]

File: packages/generator/src/storybook.ts (hunk 2 above)
Comment (rahul1) [reply packages/generator/src/storybook.ts:23]: I think they're needed for these stories: 
https://github.com/medplum/medplum/blob/main/packages/react/src/stories/ResourceTable.stories.tsx#L43-L47
https://github.com/medplum/medplum/blob/main/packages/react/src/stories/ResourceTable.stories.tsx#L43-L47

Comment (rahul1) [reply packages/generator/src/storybook.ts:23]: I may have had a merge error on my previous PR that removed them from structuredefinitions.json.

Comment (rahul1) [reply packages/generator/src/storybook.ts:23]: @codyebberson can you just check that those 2 stories work before merging?

**Review by rahul1 — APPROVED** [review]

**Review by rahul1 — CHANGES_REQUESTED** [review]
@codyebberson could you add those structure definitions back in? It seems like it caused some stories to break. 

https://medplum-storybook-git-cody-fix-scheduler-medplum.vercel.app/?path=/docs/medplum-resourcetable--covid-19-specimen-requirement

**Pull request merged**